src/main/webapp/app/components/diff-viewer/DiffViewer.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import DiffMatchPatch from 'diff-match-patch';
+import Tabs from 'app/components/tabs/tabs';
+import { Input } from 'reactstrap';
+import classnames from 'classnames';
+import * as styles from './style.module.scss';
+import { RealtimeTextAreaInput } from 'app/shared/firebase/input/RealtimeInputs';
+import { Database, onValue, ref } from 'firebase/database';
+
+export type FirebaseContent = {
+  path: string;
+  db: Database;
+};
+type DiffViewerProps = {
+  new: string | FirebaseContent;
+  old: string;
+  type: 'tabs' | 'stack' | 'merged';
+  className?: string;
+};
+
+const getMergedDiff = (newContent: string, oldContent: string) => {
+  const dmp = new DiffMatchPatch();
+  const diff = dmp.diff_main(oldContent, newContent);
+  dmp.diff_cleanupSemantic(diff);
+  return dmp.diff_prettyHtml(diff);
+};
+
+const TabsDiff = (props: DiffViewerProps) => {
+  const [inputValue, setInputValue] = useState('');
+  useEffect(() => {
+    const callbacks = [];
+
+    if (typeof props.new !== 'string') {
+      callbacks.push(
+        onValue(ref(props.new.db, props.new.path), snapshot => {
+          setInputValue(snapshot.val());
+        }),
+      );
+    } else {
+      setInputValue(props.new);
+    }
+    return () => {
+      callbacks.forEach(callback => callback?.());
+    };
+  }, [props.new]);
+
+  const diffHtml = useMemo(() => {
+    return getMergedDiff(inputValue, props.old);
+  }, [inputValue]);
+
+  return (
+    <Tabs
+      className={'mx-0'}
+      tabs={[
+        {
+          title: 'New',
+          content: (
+            <div>
+              {typeof props.new === 'string' && <Input type={'textarea'} value={props.new} style={{ backgroundColor: '#f1f9ff80' }} />}
```

Comment:
In the case where `props.new === 'string`, I assume the intention is just to display the data and not allow editing. Maybe we could set 'disabled' for the input field or just us a span instead?

Reply:
Agree. I will play around with span vs disabled.

Reply:
I decided to do a disabled textarea instead of showing span. Using span makes the content section and diff section to similar

 
<img width="1545" alt="Screenshot 2024-06-12 at 8 29 56 AM" src="https://github.com/oncokb/oncokb-transcript/assets/5400599/2560545d-954a-41ee-b43b-4dc3e515fef0">
<img width="1512" alt="Screenshot 2024-06-12 at 8 30 23 AM" src="https://github.com/oncokb/oncokb-transcript/assets/5400599/aaafdc73-efe8-4062-bcb5-2cf07620d901">